[Question] PrizmLet директива не вычисляет тип корректно в шаблоне(А17, typescript 5.4.5)

### Библиотека
- [x] `@prizm-ui/helpers`

### Компонент
PrizmLetDirective

### Версия
NodeJS:

```
$ node -v
v16.14.2
```

Dependencies:
````
"@angular-devkit/build-angular": "17.3.4",
"@angular-devkit/core": "17.3.4",
"@angular-devkit/schematics": "17.3.4",
"@angular-eslint/eslint-plugin": "17.3.0",
"@angular-eslint/eslint-plugin-template": "17.3.0",
"@angular-eslint/template-parser": "17.3.0",
"@angular-extensions/svg-icons-builder": "^11.0.0",
"@angular/cli": "~17.3.0",
"@angular-builders/custom-webpack": "^17.0.2",
"@angular-material-components/datetime-picker": "^16.0.1",
"@angular-material-components/moment-adapter": "^16.0.1",
"@angular/animations": "17.3.4",
"@angular/cdk": "16.2.14",
"@angular/cdk-experimental": "16.2.14",
````

Prizm 4.3.5
```
"@prizm-ui/components": "~4.3.4",
"@prizm-ui/core": "~4.3.4",
"@prizm-ui/helpers": "~4.3.4",
"@prizm-ui/i18n": "~4.3.4",
"@prizm-ui/icons": "~4.3.4",
"@prizm-ui/theme": "~4.3.4",
```

### Проект в котором используется Prizm
IDP

### Вопрос
В нашем проекте используется А17 и Typescript версии 5.4.5. При использовании директивы prizmLet не в булевом контексте тип не вычисляется. Ниже пример кода:

Template:
````
<ng-container *evjLet="measures$ | async as measures">
  <cdk-accordion-item class="measures" #accordionItem="cdkAccordionItem">
    <event-measure-card
      *ngFor="let measure of measures | slice: 0:accordionItemsAmount"
      (openEditorModal)="openEditorModal(measure)"
      [measure]="measure"></event-measure-card>
    <div class="measures" *ngIf="accordionItem.expanded" @slideInOutTop>
      <event-measure-card
        *ngFor="let measure of measures | slice: accordionItemsAmount"
        (openEditorModal)="openEditorModal(measure)"
        [measure]="measure"></event-measure-card>
    </div>
  </cdk-accordion-item>
````

Model: 
````
public measures$: BehaviorSubject<IEventMeasure[]> = new BehaviorSubject<IEventMeasure[]>([]);
````

Error:
````
Error: apps/ejco/src/app/widgets/evj-events-workspace/shared/components/event-measures/event-measures.component.html:19:42 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'IEventMeasure'.

19       (openEditorModal)="openEditorModal(measure)"
                                            ~~~~~~~

  apps/ejco/src/app/widgets/evj-events-workspace/shared/components/event-measures/event-measures.component.ts:23:16
    23   templateUrl: './event-measures.component.html',
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component EventMeasuresComponent.


Error: apps/ejco/src/app/widgets/evj-events-workspace/shared/components/event-measures/event-measures.component.html:20:8 - error TS2740: Type '{}' is missing the following properties from type 'IEventMeasure': isEditable, canUpdateNameAndResponsible, id, name, and 4 more.

20       [measure]="measure"></event-measure-card>
          ~~~~~~~

  apps/ejco/src/app/widgets/evj-events-workspace/shared/components/event-measures/event-measures.component.ts:23:16
    23   templateUrl: './event-measures.component.html',
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component EventMeasuresComponent.


Error: apps/ejco/src/app/widgets/evj-events-workspace/shared/components/event-measures/event-measures.component.html:24:44 - error TS2345: Argument of type 'unknown' is not assignable to parameter of type 'IEventMeasure'.

24         (openEditorModal)="openEditorModal(measure)"
                                              ~~~~~~~

  apps/ejco/src/app/widgets/evj-events-workspace/shared/components/event-measures/event-measures.component.ts:23:16
    23   templateUrl: './event-measures.component.html',
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component EventMeasuresComponent.


Error: apps/ejco/src/app/widgets/evj-events-workspace/shared/components/event-measures/event-measures.component.html:25:10 - error TS2322: Type 'unknown' is not assignable to type 'IEventMeasure'.

25         [measure]="measure"></event-measure-card>
            ~~~~~~~

  apps/ejco/src/app/widgets/evj-events-workspace/shared/components/event-measures/event-measures.component.ts:23:16
    23   templateUrl: './event-measures.component.html',
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component EventMeasuresComponent.

× Failed to compile.


````

В другом проекте при TypeScript версии 5.2.2 и А16 всё работает как ожидается

--- a/libs/helpers/src/lib/directives/let/let-context.service.ts
+++ b/libs/helpers/src/lib/directives/let/let-context.service.ts
@@ -1,6 +1,10 @@
 import { Injectable, OnDestroy } from '@angular/core';
 import { BehaviorSubject, Observable } from 'rxjs';
 
+/**
+ * @deprecated
+ * use angular @let instead
+ */
 @Injectable()
 export class PrizmLetContextService<T> implements OnDestroy {
   private readonly context$$ = new BehaviorSubject<T | null>(null);

--- a/libs/helpers/src/lib/directives/let/let.directive.ts
+++ b/libs/helpers/src/lib/directives/let/let.directive.ts
@@ -14,6 +14,11 @@ interface LetContext<T> {
  * @button <ng-container *prizmLet="{items: items$ | async, center: center} as $"> {{$.items?.count}} {{$.center}}</ng-container>
  * @button <ng-container *prizmLet="queryParams.isMap$ | async as isMap">{{isMap}}</ng-container>
  */
+
+/**
+ * @deprecated
+ * use angular @let instead
+ */
 @Directive({
   selector: '[prizmLet]',
   exportAs: 'prizmLet',

